### ISSUE TITLE
bioscrape fails to compile using installation directions

### ISSUE THREAD

**Issue by murrayrm** [ananswam/bioscrape#26]
I tried followed the directions [here](https://github.com/ananswam/bioscrape/wiki/Installation) on MacOS 10.14.2 and received the following error:
```
(buildacell) nemaya:bioscrape murray$ python setup.py install
Installing Bioscrape
/Users/murray/anaconda3/envs/buildacell/lib/python3.6/distutils/dist.py:261: UserWarning: Unknown distribution option: 'zip_safe'
  warnings.warn(msg)
running install
running build
running build_py
running build_ext
building 'bioscrape.random' extension
gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/Users/murray/anaconda3/envs/buildacell/include -arch x86_64 -I/Users/murray/anaconda3/envs/buildacell/include -arch x86_64 -I/Users/murray/anaconda3/envs/buildacell/lib/python3.6/site-packages/numpy/core/include -I. -I/Users/murray/anaconda3/envs/buildacell/include/python3.6m -c bioscrape/random.cpp -o build/temp.macosx-10.7-x86_64-3.6/bioscrape/random.o
warning: include path for stdlibc++ headers not found; pass '-std=libc++' on the
      command line to use the libc++ standard library instead
      [-Wstdlibcxx-not-found]
In file included from bioscrape/random.cpp:650:
In file included from /Users/murray/anaconda3/envs/buildacell/lib/python3.6/site-packages/numpy/core/include/numpy/arrayobject.h:4:
In file included from /Users/murray/anaconda3/envs/buildacell/lib/python3.6/site-packages/numpy/core/include/numpy/ndarrayobject.h:18:
In file included from /Users/murray/anaconda3/envs/buildacell/lib/python3.6/site-packages/numpy/core/include/numpy/ndarraytypes.h:1823:
/Users/murray/anaconda3/envs/buildacell/lib/python3.6/site-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:15:2: warning: 
      "Using deprecated NumPy API, disable it by "          "#defining
      NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-W#warnings]
#warning "Using deprecated NumPy API, disable it by " \
 ^
bioscrape/random.cpp:869:14: fatal error: 'complex' file not found
    #include <complex>
             ^~~~~~~~~
2 warnings and 1 error generated.
error: command 'gcc' failed with exit status 1
Installing Bioscrape Lineages
running install
running build
running build_py
package init file 'lineage/__init__.py' not found (or not a regular file)
package init file 'lineage/__init__.py' not found (or not a regular file)
running build_ext
building 'bioscrape.lineage' extension
gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/Users/murray/anaconda3/envs/buildacell/include -arch x86_64 -I/Users/murray/anaconda3/envs/buildacell/include -arch x86_64 -I/Users/murray/anaconda3/envs/buildacell/lib/python3.6/site-packages/numpy/core/include -I. -I/Users/murray/anaconda3/envs/buildacell/include/python3.6m -c lineage/lineage.cpp -o build/temp.macosx-10.7-x86_64-3.6/lineage/lineage.o
warning: include path for stdlibc++ headers not found; pass '-std=libc++' on the
      command line to use the libc++ standard library instead
      [-Wstdlibcxx-not-found]
In file included from lineage/lineage.cpp:648:
In file included from /Users/murray/anaconda3/envs/buildacell/lib/python3.6/site-packages/numpy/core/include/numpy/arrayobject.h:4:
In file included from /Users/murray/anaconda3/envs/buildacell/lib/python3.6/site-packages/numpy/core/include/numpy/ndarrayobject.h:18:
In file included from /Users/murray/anaconda3/envs/buildacell/lib/python3.6/site-packages/numpy/core/include/numpy/ndarraytypes.h:1823:
/Users/murray/anaconda3/envs/buildacell/lib/python3.6/site-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:15:2: warning: 
      "Using deprecated NumPy API, disable it by "          "#defining
      NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-W#warnings]
#warning "Using deprecated NumPy API, disable it by " \
 ^
lineage/lineage.cpp:650:10: fatal error: 'vector' file not found
#include <vector>
         ^~~~~~~~
2 warnings and 1 error generated.
error: command 'gcc' failed with exit status 1
```

**Comment by ananswam** [comment]
Looks like the C++ library is not being found properly.

1. Do you have XCode installed? Looks like yes since gcc runs properly.
2. Is your Cython up to date? Potentially try updating cython and numpy and
then re-attempting.
3. Can try pulling the release
https://github.com/ananswam/bioscrape/releases/tag/v0.0.1 and see if that
installs.

Build works for me running:
python setup_bioscrape.py install

using macOS 10.13.5 with Cython 0.26.1, numpy 1.14.2

**Comment by ananswam** [comment]
https://github.com/cython/cython/issues/2694 looks relevant as well, might require updating XCode

**Comment by murrayrm** [comment]
Xcode version: 
```
package-id: com.apple.pkg.CLTools_Executables
version: 10.1.0.0.1.1539992718
volume: /
location: /
install-time: 1549658277
groups: com.apple.FindSystemFiles.pkg-group 
```

Python package versions:
```
# packages in environment at /Users/murray/anaconda3/envs/buildacell:
#
# Name                    Version                   Build  Channel
beautifulsoup4            4.7.1                    py36_1    anaconda
blas                      1.0                         mkl    anaconda
ca-certificates           2019.1.23                     0    anaconda
certifi                   2018.11.29               py36_0    anaconda
cython                    0.29.4           py36h0a44026_0    anaconda
emcee                     2.2.1                    pypi_0    pypi
icu                       58.2                 h4b95b61_1    anaconda
intel-openmp              2019.1                      144    anaconda
libcxx                    4.0.1                hcfea43d_1  
libcxxabi                 4.0.1                hcfea43d_1  
libedit                   3.1.20181209         hb402a30_0  
libffi                    3.2.1                h475c297_4  
libgfortran               3.0.1                h93005f0_2    anaconda
libiconv                  1.15                 hdd342a3_7    anaconda
libxml2                   2.9.9                hab757c2_0    anaconda
libxslt                   1.1.33               h33a18ac_0    anaconda
lxml                      4.3.0            py36hef8c89e_0    anaconda
mkl                       2019.1                      144    anaconda
mkl_fft                   1.0.10           py36h5e564d8_0    anaconda
mkl_random                1.0.2            py36h27c97d8_0    anaconda
ncurses                   6.1                  h0a44026_1  
numpy                     1.15.4           py36hacdab7b_0    anaconda
numpy-base                1.15.4           py36h6575580_0    anaconda
openssl                   1.1.1                h1de35cc_0    anaconda
pip                       19.0.1                   py36_0  
python                    3.6.8                haf84260_0  
python-libsbml            5.17.0                   py36_0    SBMLTeam
readline                  7.0                  h1de35cc_5  
scipy                     1.2.0            py36h1410ff5_0    anaconda
setuptools                40.7.3                   py36_0  
soupsieve                 1.7.1                    py36_0    anaconda
sqlite                    3.26.0               ha441bb4_0  
tk                        8.6.8                ha441bb4_0  
wheel                     0.32.3                   py36_0  
xz                        5.2.4                h1de35cc_4  
zlib                      1.2.11               h1de35cc_3  
```

**Comment by murrayrm** [comment]
Different error when I run `python setup_bioscrape.py install`
```
/Users/murray/anaconda3/envs/buildacell/lib/python3.6/distutils/dist.py:261: UserWarning: Unknown distribution option: 'zip_safe'
  warnings.warn(msg)
running install
running build
running build_py
running build_ext
building 'bioscrape.random' extension
gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/Users/murray/anaconda3/envs/buildacell/include -arch x86_64 -I/Users/murray/anaconda3/envs/buildacell/include -arch x86_64 -I/Users/murray/anaconda3/envs/buildacell/lib/python3.6/site-packages/numpy/core/include -I. -I/Users/murray/anaconda3/envs/buildacell/include/python3.6m -c bioscrape/random.cpp -o build/temp.macosx-10.7-x86_64-3.6/bioscrape/random.o
warning: include path for stdlibc++ headers not found; pass '-std=libc++' on the
      command line to use the libc++ standard library instead
      [-Wstdlibcxx-not-found]
In file included from bioscrape/random.cpp:650:
In file included from /Users/murray/anaconda3/envs/buildacell/lib/python3.6/site-packages/numpy/core/include/numpy/arrayobject.h:4:
In file included from /Users/murray/anaconda3/envs/buildacell/lib/python3.6/site-packages/numpy/core/include/numpy/ndarrayobject.h:18:
In file included from /Users/murray/anaconda3/envs/buildacell/lib/python3.6/site-packages/numpy/core/include/numpy/ndarraytypes.h:1823:
/Users/murray/anaconda3/envs/buildacell/lib/python3.6/site-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:15:2: warning: 
      "Using deprecated NumPy API, disable it by "          "#defining
      NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-W#warnings]
#warning "Using deprecated NumPy API, disable it by " \
 ^
bioscrape/random.cpp:869:14: fatal error: 'complex' file not found
    #include <complex>
             ^~~~~~~~~
2 warnings and 1 error generated.
error: command 'gcc' failed with exit status 1
```

**Comment by ananswam** [comment]
FIxed in 4ae2edbda0 by checking platform and modifying compile and link flags appropriately in setup_bioscrape/setup_lineages.py if the user is on macOS.
